aws/offsite/lambda/verify-database-copy/index.js

```diff
@@ -0,0 +1,114 @@
+const SLACK_CHANNEL = process.env.SLACK_CHANNEL;
+
+const DB_INSTANCE_IDENTIFIER = process.env.DB_INSTANCE_IDENTIFIER;
+const DB_HOST = process.env.DB_HOST;
+const DB_NAME = process.env.DB_NAME;
+const DB_USER = process.env.DB_USER;
+const DB_PASSWORD = process.env.DB_PASSWORD;
+
+var status_message = '';
+
+
+var postStatusToSlack = function postStatusToSlack(message) {
+    console.log('Posting status to Slack');
+    var slack = new Slack();
+    slack.setWebhook(SLACK_WEBHOOK_URL);
+    slack.webhook({
+            channel: SLACK_CHANNEL,
+            text: message
+        },
+        function (error, response) {
+            if (error) {
+                console.log('Slack Post Error: ' + JSON.stringify(error));
+            } else {
+                console.log('Slack Post Response: ' + JSON.stringify(response));
+            }
+        }
+    );
+};
+
+
+exports.handler = (event, context, callback) => {
+    console.log(JSON.stringify(event));
+    var eventMessage = JSON.parse(event.Records[0].Sns.Message);
+    var eventID = eventMessage['Event ID'];
+    if (eventID.endsWith('RDS-EVENT-0002')) {
+        var rds = new AWS.RDS();
+        var mysqlConnection;
+
+        rds.modifyDBInstance({
+            DBInstanceIdentifier: DB_INSTANCE_IDENTIFIER,
+            MasterUserPassword: DB_PASSWORD
+        }).promise()
+            .then(function (data) {
+                console.log('Modify database password request has been submitted successfully.  Sleep for a few minutes until it completes.');
+                // updating an RDS database master password typically takes 2-3 minutes, during which time the database transitions through several states:
+                // 1) Available with old/unknown password.  Attempting to connect with new password will fail authentication
+                // 2) Not Available (Status = resetting-master-credentials).  Attempt to connect as a mysql client or to invoke AWS describeDBInstances API fails
+                // 3) Available with new password
+                // Wait until about 2 minutes before this Lambda function reaches its 5 minute timeout before attempting to connect to the database
+                return setTimeoutPromise(context.getRemainingTimeInMillis() - 60000, null);
+            })
+            .then(function (value) {
+                // check if password update is pending
+                return rds.describeDBInstances({
+                        DBInstanceIdentifier: 'verification-copy'
+                    }).promise();
+            })
+            .then(function (data) {
+                if (data.DBInstances[0].PendingModifiedValues.hasOwnProperty('MasterUserPassword')) {
+                    status_message = 'Terminating verification of database copy because database password change has not started yet.';
+                    console.error(status_message);
+                    postStatusToSlack(status_message);
+                    callback(new Error(status_message));
```

Comment:
This appears to duplicate what's in the `catch` block at the end. You can simplify this by replacing these 3 lines with `return Promise.reject(new Error(status_message))` or possibly even just `return Promise.reject(status_message)`.